### PR TITLE
Desired forecast length, cronjob schedule, upload realtime forecast to s3 bucket

### PR DESCRIPTION
**Pull request by SadeghTabas-NOAA**
### Description
This PR provides updates on the `run_graphcast.py` script to provide the flexibility of running graphcast for the desired forecast lead time. In addition, it provides scripts to schedule a cronjob on both cloud and hera machines. As the hera machine does not have access to the internet, we needed to break down the job to three sections including, generating inputs to graphcast, running graphcast, and uploading the forecast to s3 bucket. The first and and last sections will be run on the control node while the second one (running graphcast) will run on the compute node. Therefore a script is provided to call the upload_to_s3 function to upload the forecasts to s3 bucket.  In addition, some bugs with 'gdas_utility.py` got fixed.

### Linked Issues
- Closes https://github.com/NOAA-EMC/graphcast/issues/24

### Input data
- [x] No changes are expected to input data.
- [ ] Changes are expected to input data:
  - [ ] New input data.
  - [ ] Updated input data.

### Needed libraries

-

### CODE REVIEW

File: NCEP/cronjob_hera.sh [hunk 1]
```diff
@@ -0,0 +1,21 @@
+#!/bin/bash --login
+
+echo "Job 1 is running"
+sh /scratch1/NCEPDEV/nems/AIML/graphcast/NCEP/gc_job1_hera.sh
```
Comment (junwang-noaa) [inline NCEP/cronjob_hera.sh:4]: Can we call this: gc_prepdata_hera.sh to be more specific?

File: NCEP/cronjob_hera.sh [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+sleep 60  # Simulating some work
+echo "Job 1 completed"
+
+echo "Job 2 is running"
+job2_id=$(sbatch /scratch1/NCEPDEV/nems/AIML/graphcast/NCEP/gc_job2_hera.sh | awk '{print $4}')
```
Comment (junwang-noaa) [inline NCEP/cronjob_hera.sh:9]: Here too: maybe: gc_runfcst_hera.sh?

File: NCEP/cronjob_hera.sh [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+
+# Wait for job 2 to complete
+while squeue -j $job2_id &>/dev/null; do
+    sleep 5  # Adjust the polling interval as needed
+done
+sleep 5  # Simulating some work
+echo "Job 2 completed"
+
+echo "Job 3 is running"
+sh /scratch1/NCEPDEV/nems/AIML/graphcast/NCEP/gc_job3_hera.sh
```
Comment (junwang-noaa) [inline NCEP/cronjob_hera.sh:19]: gc_datadissm_hera.sh?

**Review by junwang-noaa — APPROVED** [review]
The code looks good, we can change the job names later.

**Comment by SadeghTabas-NOAA** [comment]
> The code looks good, we can change the job names later.

Sure, Thanks

**Comment by SadeghTabas-NOAA** [comment]
> The code looks good, we can change the job names later.

I did update the names. Thanks

File: NCEP/cronjob_hera.sh (hunk 3 above)
Comment (SadeghTabas-NOAA) [reply NCEP/cronjob_hera.sh]: updated

File: NCEP/cronjob_hera.sh (hunk 2 above)
Comment (SadeghTabas-NOAA) [reply NCEP/cronjob_hera.sh]: updated

File: NCEP/cronjob_hera.sh (hunk 1 above)
Comment (SadeghTabas-NOAA) [reply NCEP/cronjob_hera.sh]: updated

**Review by junwang-noaa — APPROVED** [review]

**Pull request merged**